DownloadManager/Plugin.DownloadManager.Android/DownloadFileImplementation.cs

```diff
@@ -92,20 +94,44 @@ public DownloadFileImplementation (ICursor cursor)
             Url = cursor.GetString (cursor.GetColumnIndex (Android.App.DownloadManager.ColumnUri));
         }
 
-        public void StartDownload (Android.App.DownloadManager downloadManager, string destinationPathName)
+        private bool _mobileNetworkAllowed = true;
+
+        public bool MobileNetworkAllowed
         {
-            using (var downloadUrl = Uri.Parse (Url))
-            using (var request = new Android.App.DownloadManager.Request (downloadUrl)) {
+            get
+            {
+                return _mobileNetworkAllowed;
+            }
+            set
+            {
+                _mobileNetworkAllowed = value;
+                if (Request != null && value == false) {
+                    Request.SetAllowedNetworkTypes(Android.App.DownloadNetwork.Wifi);
```

Comment:
@martijn00 Why should we go for v24, when the official documentation of Xamarin tells me to avoid it? https://developer.xamarin.com/guides/android/application_fundamentals/understanding_android_api_levels/#API_Levels_and_Libraries